### PR TITLE
Add license checking

### PR DESCRIPTION
**Pull request by Ivan-Velickovic**
Following libvmm, sDDF and seL4 Foundation projects.

### CODE REVIEW

File: .github/workflows/push.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  check:
+    name: License Check
+    runs-on: ubuntu-latest
+    steps:
+    - uses: seL4/ci-actions/license-check@master
```
Comment (wom-bat) [inline .github/workflows/push.yaml:19]: I'm uncomfortable about using checks from repos in a different Github organisation.  Right now this is probably OK, as seL4 is 'close enough' to us; but it'd be better to have our own au-ts checks in the future.

File: components/fs/nfs/lwip_include/lwipopts.h [hunk 2]
```diff
@@ -1,13 +1,6 @@
...
- * Copyright 2020, Data61
- * Commonwealth Scientific and Industrial Research Organisation (CSIRO)
- * ABN 41 687 119 230.
- *
- * This software may be distributed and modified according to the terms of
- * the BSD 2-Clause license. Note that NO WARRANTY is provided.
- * See "LICENSE_BSD2.txt" for details.
- *
- * @TAG(DATA61_BSD)
+ * Copyright 2020, Data61, CSIRO
```
Comment (wom-bat) [inline components/fs/nfs/lwip_include/lwipopts.h:2]: The previous copyright notice was checked by Data61's lawyers.  I think you need to keep the full name of CSIRO, and the ABN.

File: examples/kitty/board/odroidc4/kitty.system [hunk 3]
```diff
@@ -1,4 +1,9 @@
 <?xml version="1.0" encoding="UTF-8"?>
+<!--
+    Copyright 2024, UNSW
+
```
Comment (wom-bat) [inline examples/kitty/board/odroidc4/kitty.system:4]: Extra space?

**Review by wom-bat — APPROVED** [review]
LGTM

**Comment by Ivan-Velickovic** [comment]
Addressed comments + rebased.

File: .github/workflows/push.yaml (hunk 1 above)
Comment (Ivan-Velickovic) [reply .github/workflows/push.yaml:19]: I agree but thought it was okay in this case since it's the Foundation.

**Pull request merged**